api/wallet_test.go

```diff
@@ -1474,3 +1474,158 @@ func TestWalletGETDust(t *testing.T) {
 		t.Fatal("dustThreshold mismatch")
 	}
 }
+
+// testWalletTransactionEndpoint is a subtest that queries the transaction endpoint of a node.
+func testWalletTransactionEndpoint(t *testing.T, st *serverTester) {
+	// Mining blocks should have created transactions for the wallet containing
+	// miner payouts. Get the list of transactions.
+	var wtg WalletTransactionsGET
+	err := st.getAPI("/wallet/transactions?startheight=0&endheight=-1", &wtg)
+	if err != nil {
+		t.Fatal(err)
+	}
```

Comment:
Should have a check here that the number of transactions matches the number of expected transactions.

Reply:
That may mean that you have to pass the number of expected transactions in as input.